src/artemis/experiment_plans/stepped_grid_scan_plan.py

```diff
@@ -0,0 +1,199 @@
+from __future__ import annotations
+
+import argparse
+from typing import TYPE_CHECKING, Callable
+
+import bluesky.plan_stubs as bps
+from bluesky import RunEngine
+from bluesky.plans import grid_scan
+from bluesky.utils import ProgressBarManager
+from dodal import i03
+from dodal.devices.aperturescatterguard import AperturePositions
+from dodal.devices.eiger import DetectorParams
+from dodal.i03 import Smargon
+
+from artemis.log import LOGGER
+from artemis.parameters import external_parameters
+from artemis.parameters.beamline_parameters import (
+    GDABeamlineParameters,
+    get_beamline_prefixes,
+)
+from artemis.parameters.constants import I03_BEAMLINE_PARAMETER_PATH, SIM_BEAMLINE
+from artemis.tracing import TRACER
+from artemis.utils.utils import Point3D
+
+if TYPE_CHECKING:
+    from artemis.external_interaction.callbacks.stepped_grid_scan.stepped_grid_scan_callback_collection import (
+        SteppedGridScanCallbackCollection,
+    )
+    from artemis.parameters.internal_parameters.plan_specific.stepped_grid_scan_internal_params import (
+        SteppedGridScanInternalParameters,
+    )
+
+
+class SteppedGridScanComposite:
+    """A device consisting of all the Devices required for a stepped grid scan."""
+
+    sample_motors: Smargon
+
+    def __init__(
+        self,
+        aperture_positions: AperturePositions = None,
+        detector_params: DetectorParams = None,
+        fake: bool = False,
+    ):
+        self.sample_motors = i03.smargon(fake_with_ophyd_sim=fake)
+
+
+stepped_grid_scan_composite: SteppedGridScanComposite | None = None
+
+
+def get_beamline_parameters():
+    return GDABeamlineParameters.from_file(I03_BEAMLINE_PARAMETER_PATH)
+
+
+def create_devices():
+    """Creates the devices required for the plan and connect to them"""
+    global stepped_grid_scan_composite
+    prefixes = get_beamline_prefixes()
+    LOGGER.info(
+        f"Creating devices for {prefixes.beamline_prefix} and {prefixes.insertion_prefix}"
+    )
+    aperture_positions = AperturePositions.from_gda_beamline_params(
+        get_beamline_parameters()
+    )
+    LOGGER.info("Connecting to EPICS devices...")
+    stepped_grid_scan_composite = SteppedGridScanComposite(
+        aperture_positions=aperture_positions
+    )
+    LOGGER.info("Connected.")
+
+
+def move_xyz(
+    sample_motors,
+    xray_centre_motor_position: Point3D,
+    md={
+        "plan_name": "move_xyz",
+    },
+):
+    """Move 'sample motors' to a specific motor position (e.g. a position obtained
+    from gridscan processing results)"""
+    LOGGER.info(f"Moving Smargon x, y, z to: {xray_centre_motor_position}")
+    yield from bps.mv(
+        sample_motors.x,
+        xray_centre_motor_position.x,
+        sample_motors.y,
+        xray_centre_motor_position.y,
+        sample_motors.z,
+        xray_centre_motor_position.z,
+    )
```

Comment:
Should: Maybe a new issue but this is an exact copy of the `fast_grid_scan` one, it should probably be somewhere common